### PR TITLE
Remove skip evaluation of symlinks to data root on IoT Core

### PR DESCRIPTION
**Pull request by thaJeztah**
This fix was added in https://github.com/moby/moby/pull/33498/commits/8e71b1e210dc0eff980f39271d6c1dd48d87024e (https://github.com/moby/moby/pull/33498) to work around a go issue (https://github.com/golang/go/issues/20506).

That issue was fixed in https://github.com/golang/go/commit/66c03d39f3aa65ec522c41e56c569391786539a7,
which is part of Go 1.10 and up. This reverts the changes that were made in
8e71b1e210dc0eff980f39271d6c1dd48d87024e, and are no longer needed.

### CODE REVIEW

File: pkg/system/syscall_windows.go
Comment (thaJeztah) [inline pkg/system/syscall_windows.go]: Looks like this function is no longer used in our code, so perhaps we can remove it altogether (a quick search on GitHub also didn't show any users of this)

**Comment by thaJeztah** [comment]
ping @jhowardmsft @darstahl PTAL 🤗

**Review by feJustino — APPROVED** [review]

**Comment by jhowardmsft** [comment]
@taylorb-microsoft Can you loop in one of the IoT guys to verify. I believe this is OK, but I'd like their verification first.

File: pkg/system/syscall_windows.go
Comment (jhowardmsft) [reply pkg/system/syscall_windows.go]: Yes, this is the last user of this function after #38000

**Comment by thaJeztah** [comment]
Thanks! Yes; this was purely based on that comment, so would be good to double check 🤗

**Comment by darstahl** [comment]
I no longer have an environment to test this, but LGTM pending verification that it works as intended on an IoT build.

**Comment by jhowardmsft** [comment]
Hold fire on this - internal testing is throwing an issue. It's being investigated.

**Comment by jhowardmsft** [comment]
@jiria Following on from our discussion yesterday, are you OK with this? I think it's safe now, but if you can confirm.

**Comment by jiria** [comment]
@jhowardmsft looks good to me.

**Comment by thaJeztah** [comment]
rebased, and removed "WIP"

**Comment by olljanat** [comment]
Derek add label: rebuild/*

**Comment by olljanat** [comment]
Derek add label: rebuild/powerpc

**Comment by thaJeztah** [comment]
rebased again; @jhowardmsft @kolyshkin PTAL

**Comment by thaJeztah** [comment]
Experimental failure looks to be a flaky; https://jenkins.dockerproject.org/job/Docker-PRs-experimental/45949/console

```
11:18:38 FAIL: docker_api_swarm_node_test.go:75: DockerSwarmSuite.TestAPISwarmNodeDrainPause
11:18:38 
11:18:38 Creating a new daemon
11:18:38 [dc139d4dd25da] waiting for daemon to start
11:18:38 [dc139d4dd25da] waiting for daemon to start
11:18:38 [dc139d4dd25da] daemon started
11:18:38 
11:18:38 Creating a new daemon
11:18:38 [d53117bd699ed] waiting for daemon to start
11:18:38 [d53117bd699ed] waiting for daemon to start
11:18:38 [d53117bd699ed] daemon started
11:18:38 
11:18:38 waited for 707.626002ms (out of 30s)
11:18:38 waited for 9.542429ms (out of 30s)
11:18:38 waited for 18.527488ms (out of 30s)
11:18:38 waited for 2.227709728s (out of 30s)
11:18:38 waited for 8.748001ms (out of 30s)
11:18:38 docker_api_swarm_node_test.go:103:
11:18:38     waitAndAssert(c, defaultReconciliationTimeout*2, reducedCheck(sumAsIntegers, d1.CheckActiveContainerCount, d2.CheckActiveContainerCount), checker.Equals, instances)
11:18:38 docker_utils_test.go:435:
11:18:38     c.Assert(v, checker, args...)
11:18:38 ... obtained int = 2
11:18:38 ... expected int = 1
11:18:38 ... output: "801028f0f936\ne5772591d4b2\n"
11:18:38 
11:18:38 waited for 1m0.063567148s (out of 1m0s)
11:18:38 [dc139d4dd25da] Stopping daemon
11:18:38 [dc139d4dd25da] exiting daemon
11:18:38 [dc139d4dd25da] Daemon stopped
11:18:38 [d53117bd699ed] Stopping daemon
11:18:38 [d53117bd699ed] exiting daemon
11:18:38 [d53117bd699ed] Daemon stopped
```

Daemon logs attached:

[d53117bd699ed.log](https://github.com/moby/moby/files/3389990/d53117bd699ed.log)
[dc139d4dd25da.log](https://github.com/moby/moby/files/3389991/dc139d4dd25da.log)

**Comment by thaJeztah** [comment]
ok, this is rebased and green

@jhowardmsft @cpuguy83 @AkihiroSuda PTAL

**Review by lowenna — APPROVED** [review]
LGTM @jiria FYI

**Review by cpuguy83 — APPROVED** [review]
LGTM

**Pull request merged**
